src/dotnet/CarbonAware.WebApi/Controllers/SciScoreController.cs

```diff
@@ -1,45 +1,141 @@
+                _logger.LogError("get carbon intensity failed with error: {error}", error);
+                return BadRequest(error);
+            }
+
+            // check that there is a time interval passed in
+            if (String.IsNullOrEmpty(input.TimeInterval))
             {
-                SciScoreValue = 100.0f,
-                EnergyValue = 1.0f,
-                MarginalCarbonEmissionsValue = 100.0f,
-                EmbodiedEmissionsValue = 0.0f,
-                FunctionalUnitValue = 1
+                var error = new CarbonAwareWebApiError() { Message = "TimeInterval is required" };
+                _logger.LogError("get carbon intensity failed with error: {error}", error);
+                return BadRequest(error);
+            }
+            try
+            {
+                var carbonIntensity = await _aggregator.CalculateAverageCarbonIntensityAsync(GetLocation(input.Location), input.TimeInterval);
+
+                SciScore score = new SciScore
+                {
+                    MarginalCarbonIntensityValue = carbonIntensity,
+                };
+                _logger.LogDebug("calculated marginal carbon intensity: {score}", score);
+                return Ok(score);
+            }
+            // Catch ArgumentException for invalid inputs and broadly 3rd Party dependency exceptions
+            catch (Exception ex)
+            {
+                _logger.LogError("Exception occured during marginal calculation execution", ex);
+                var error = new CarbonAwareWebApiError() { Message = ex.ToString() };
+                return BadRequest(error);
+            }
+        }
+    }
+
+    /// Validate the user input location and convert it to the internal Location object.
+    /// Throws ArgumentException if the location input is invalid.
+    private Location GetLocation(LocationInput locationInput)
+    {
+        LocationType locationType;
+        CloudProvider cloudProvider;
+
+        if (!Enum.TryParse<LocationType>(locationInput.LocationType, true, out locationType))
+        {
+            throw new ArgumentException($"locationType '{locationInput.LocationType}' is invalid");
+        }
+
+        Enum.TryParse<CloudProvider>(locationInput.CloudProvider, true, out cloudProvider);
+
+        try
+        {
+            var location = new Location
+            {
+                LocationType = locationType,
+                Latitude = locationInput.Latitude,
+                Longitude = locationInput.Longitude,
+                CloudProvider = cloudProvider,
+                RegionName = locationInput.RegionName
             };
 
-        return await Task.Run(() => Ok(score));
+            return location;
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError("Exception occured during location creation", ex);
+            throw new ArgumentException("location provided is invalid");
```

Comment:
Only create a new exception if you're going to add additional information, like the location.